### PR TITLE
Fetch a todo item feature

### PR DESCRIPTION
**Pull request by coder4097**
## Description

Fetch a TODO item:
URI: GET /todos/:id
res.status: 200
res: { id: id, title: title }

### Database schema changes
None

### Automated test cases added

- Test in which we get response 200 if todo item exists 
- Test in which we get response 404, if todo item not found
-  Test in which we get response 400  if invalid mongo id is passed

### Manual test cases run

Done on postman

### CODE REVIEW

**Comment by coder1033** [comment]
Validate the mongoID in validators.

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "lint": "tslint --fix -c tslint.json src/**/*.ts test/**/*.ts",
     "heroku-postbuild": "npm run build",
     "heroku-start": "node ./dist/server/server",
-    "test": "NODE_ENV=testing mocha --exit -r ts-node/register --project tsconfig.json test/**/*.spec.ts",
+    "test": "mocha 'test/**/*.spec.ts' --timeout 10000 --exit -r ts-node/register --project tsconfig.json test/**/*.spec.ts",
```
Comment (coder1033) [inline package.json:12]: Why there is a need of this `'test/**/*.spec.ts' --timeout 10000`.

File: test/spec/todo/todo-item.spec.ts [hunk 2]
```diff
@@ -23,7 +24,7 @@ before(async () => {
   expressApp = app.expressApp;
 });
 
-describe('POST /todos', () => {
+describe('POST/todos', () => {
```
Comment (coder1033) [inline test/spec/todo/todo-item.spec.ts:27]: Keep the spacing `POST /todos`

File: package.json (hunk 1 above)
Comment (coder4097) [reply package.json:12]: its because my system is taking time to process promises..it was some promise error while automated test cases

Comment (coder1033) [reply package.json:12]: You should resolve that error then.

**Review by coder1033 — APPROVED** [review]

**Review by coder1033 — APPROVED** [review]

**Pull request merged**